### PR TITLE
ggplot output graph bug fixes

### LINKED ISSUE #4: gglot2 graph plot issue.
After, running the ` Rperform::plot_branchmetrics(test_path = "tests/testthat/test-CRAN.R", metric = "time", branch1 = "l1loss", branch2 = "master", save_data = T, save_plots = F)` function on the [binsegRcpp](https://github.com/tdhock/binsegRcpp) repository, comparing the **master** and **l1loss** branch. I recorded a time_frame of 254 observations and 8 variables.
Below is how the graph looks like :

Rplot

You could see that information is not depicted clearly on the graph.
My hypothesis is that it was caused as a result of the larger data set.
Potential fix will be breaking down the data sets and plotting their respective graphs.

### PR DESCRIPTION
**Pull request by EngineerDanny**
This PR fixes #4 
I have been performing several tests on the various data sets to address some of the issues with the graph output. 
More updates will follow.

### CODE REVIEW

**Comment by EngineerDanny** [comment]
**ISSUE** : When commit messages are same, the graph displays their metric values on the same x-axis coordinate. This leads to more than 3 points on the same axis. It leads to wrong interpretation of test output. In the code, the x axis uses the commit message variable whilst the y axis uses the metric value. I also noticed that due to this issue, the x-axis tick labels were random. It has to follow a particular discrete order. 

**FIX** :  Making the x-variable input unique. Replacing the `commit message` with the `commit sha`  eliminates this bug. However, since the `commit sha` contains random alphanumeric characters, it is not easily readable on the x-axis. I have fixed this issue by using the commit messages as the x axis tick labels with `ggplot2::scale_x_discrete(labels= test_data$message)`. This fix addresses comment 2 of the feedback given [here](https://github.com/EngineerDanny/stringr/pull/4#issuecomment-1183644990) .

**Comment by EngineerDanny** [comment]
@tdhock  You raised some User Interface (UI) issues with the graph [here](https://github.com/EngineerDanny/Rperform/pull/7#issuecomment-1176492016)

I have updated the code to address what you said. I have : 

1. Reverted to the -90 for the x-axis tick label angle because it looked more professional and it is preferred especially when dealing with larger data sets.
2. Adjusted the hjust and vjust of the x tick label.
3. Centered the main title and made the font bold.

This is what the output looks like now :
Screenshot 2022-07-19 at 4 09 51 PM

**Comment by tdhock** [comment]
great that is a lot better. can you use scale_y_log10 please so that large values are de-emphasized?

**Comment by EngineerDanny** [comment]
@tdhock Sure. This is how it looks with `scale_y_log10`
I must say, this is my first time using `scale_y_log10`.  I notice a slight change which makes the graph better.

Screenshot 2022-07-19 at 6 56 51 PM

**Comment by EngineerDanny** [comment]
@tdhock This is how larger datasets look like. 
Should I maintain it like this (where the whole graph is seen on the go) or make it scrollable especially on the horizontal axis ?

ggsave1

**Comment by tdhock** [comment]
This looks good. I put it full screen and it is readable. we can see clearly that some commits result in slowdowns but overall there is no big change from the beginning to the end.

**Comment by EngineerDanny** [comment]
Yes. I will perform tests on other packages.
This is a good start I guess.

**Comment by EngineerDanny** [comment]
Update comment everytime there is a push on a PR.

exposable some parameters  like the :
total_height_in
total_width_in
width_in_per_commit
height_in_per_test
res

draw_x_axis_on_each_panel = true

**Comment by EngineerDanny** [comment]
@tdhock There is this problem which has been troubling me. After I added the parameters, I needed to update the documentation so that I can generate the Rd files for the function. I have updated the documentation in the `plot_metrics.R` file but the `usage` block in the `plot_metrics.Rd` file doesn't update when I run `devtools::document()` function. 
This one below in `man/plot_metrics.Rd`, however I change the `plot_metrics` function, it doesn't update. Is there any trick I am missing?
```r
\usage{
    plot_metrics(
      test_path,
      metric,
      num_commits = 5,
      save_data = FALSE,
      save_plots = FALSE,
      interactive = FALSE
    )
}
```

**Comment by tdhock** [comment]
I'm not sure, maybe try update.packages()

**Comment by EngineerDanny** [comment]
The problem was with the encoding type. The warning message was `roxygen2 requires Encoding: UTF-8`, so from this [answer](https://stackoverflow.com/questions/51694929/warning-about-utf-8-with-roxygen2), I added  `Encoding: UTF-8` to the **DESCRIPTION** file. It fixed the issue.

**Comment by EngineerDanny** [comment]
@tdhock  From your comments, I have realised that when the `height_in_inch`, `width_in_inch` and `resolution` are adjusted well, every plot will clearly be seen. So exposing those parameters will fix the issue of plotting large datasets. Maybe in the future, I will expose other parameters like the `x-tick size` and the `geom point size`.

**Comment by EngineerDanny** [comment]
@tdhock Thanks for your comment. Replacing `the  facet_grid` with `facet_wrap(scales="free", ncol=1)` certainly works for the `draw_x_axis_on_each_panel = true`  feature. 
I am closing this PR for now. 
I will open new ones to address some of the problems I encountered while running Rperform on the other popular packages.

File: R/plot_metrics.R [hunk 1]
```diff
@@ -62,10 +67,12 @@ utils::globalVariables(c("metric_val", "test_name", "mid_val",
...
 #'   Function assumes the current directory to be the root directory of the
 #'   repository/package being tested.
+#' @name plot_metrics
 #'
 
-plot_metrics <- function(test_path, metric, num_commits = 5, save_data = FALSE, save_plots = FALSE,
-                         interactive = FALSE){                     
+plot_metrics <- function(test_path, metric, num_commits = 5,
+  save_data = FALSE, save_plots = FALSE, interactive = FALSE,
+  total_height_in, total_width_in, resolution){
```
Comment (tdhock) [inline R/plot_metrics.R:75]: defaults?

Comment (tdhock) [reply R/plot_metrics.R:75]: I think R default is width=7/height=7

**Comment by tdhock** [comment]
great looks good overall. can you add a test for the new functionality?

**Comment by EngineerDanny** [comment]
@tdhock  Yes. I have tested on some popular packages. 
I again tried to run the test on [binsegRcpp](https://github.com/tdhock/binsegRcpp). I get errors from 30 commits upward. I will investigate this issue. (Although my hypothesis is this is an internal issue with `binsegRcpp` itself) .

Screenshot 2022-08-15 at 6 25 20 PM

**Comment by EngineerDanny** [comment]
@tdhock I ran 
```r
 plot_metrics(
     test_path = "tests/testthat/test-rmd.R",
     metric = "time", num_commits = 80,
     save_data = TRUE,
     save_plots = TRUE
 )
```
on [styler](https://github.com/r-lib/styler) package, and this was the graph output :

Screenshot 2022-08-16 at 6 54 58 AM

**Pull request merged**